src/rocksdb_options.rs

```diff
@@ -319,6 +319,20 @@ impl Default for Options {
     }
 }
 
+impl Clone for Options {
+    // Only copy DBOptions.
+    fn clone(&self) -> Self {
+        unsafe {
+            let opts = crocksdb_ffi::crocksdb_options_copy(self.inner);
+            assert!(!opts.is_null());
+            Options {
+                inner: opts,
+                filter: None,
```

Comment:
looking into definition of Options, compaction_filter, comparator are all raw ptr.(while others are shared_ptr)
So that's the reason why we need a filter field in rust DBOption, (to own the mem).
So impl Clone should handle this carefully